jdbc/driver/src/main/java/io/tidb/bigdata/jdbc/impl/RandomShuffleUrlMapper.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2020 TiDB Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.tidb.bigdata.jdbc.impl;
+
+import java.util.Arrays;
+import java.util.Random;
+import java.util.function.Function;
+
+public class RandomShuffleUrlMapper implements Function<String[], String[]> {
+
+  private static final ThreadLocal<Random> threadLocalRandom = ThreadLocal.withInitial(Random::new);
```

Comment:
If we don't need to support Java 6 or older, we could directly use `java.util.concurrent.ThreadLocalRandom`.

Reply:
Good point! We are not going to support versions older than 8.

Reply:
yes, we use lambdas.